src/main/java/org/folio/entlinks/service/consortium/ConsortiumTenantExecutor.java

```diff
@@ -0,0 +1,40 @@
+package org.folio.entlinks.service.consortium;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.entlinks.client.UserTenantsClient;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor
+public class ConsortiumTenantExecutor {
+
+  private final UserTenantsClient userTenantsClient;
+  private final FolioExecutionContext folioExecutionContext;
+  private final SystemUserScopedExecutionService scopedExecutionService;
+
+  public <T> T executeAsCentralTenant(Supplier<T> operation) {
+    var tenantId = folioExecutionContext.getTenantId();
+    var centralTenantId = Optional.ofNullable(userTenantsClient.getUserTenants(tenantId))
+        .map(UserTenantsClient.UserTenants::userTenants)
+        .orElse(List.of())
+        .stream()
+        .findFirst()
+        .map(UserTenantsClient.UserTenant::centralTenantId);
```

Comment:
I think it should be cached. You can move this logic to the user tenants service and cache it.